### PR TITLE
Completed Greer combat replay and mechanics

### PR DESCRIPTION
**Pull request by Linkaaaaa**
# Greer

## Mechanics
- Fixed Noxious Blight only counting greens from the adds
- Clarified Infectious Rot description
### Added
- Hit by Rot the World
- Hit by Wave of Corruption
- Hit by Ripples of Rot
- Hit by Rain of Spores
- Hit by Enfeebling Miasma
- Hit by Scattering Sporeblast
- Hit by Aura of Corruption
- Hit by Rake the Rot
- Hit by Cage of Decay
- Hit by Sweep the Mold
- Hit by Blob of Blight
- Hit by Eruption of Rot
- Hit by Stomp the Growth
- Received Target Buff

## Combat Replay
- Fixed Noxious Blight only showing greens from the adds
- Rot the World
- Stomp the Growth
- Wave of Corruption
- Cage of Decay (Arrow + Circle indicators & various effects, they also work for Eruption of Rot)
- Ripples of Rot
- Rain of Spores
- Enfeebling Miasma (indicators + gas clouds)
- Hit by Scattering Sporeblast
- Sweep the Mold + Rake the Rot
- Blob of Blight (only the spawn aoe)

## Buffs
- Added Target buff

# General

- Fixed Combat Replay debug effect methods not being unsable (because of ifdebug the renaming of HexContentGUID never went through and couldn't reference them)
- Adjusted Dynamic Bar position for Decima's Death Charge to fit Linus' new map
- Added 2 new Decima mechanics for the new patch

### CODE REVIEW

File: GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs [hunk 1]
```diff
@@ -137,4 +234,238 @@ internal override void ComputePlayerCombatReplayActors(PlayerActor player, Parse
...
+            foreach (EffectEvent effect in miasmaIndicator)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, 6000);
+                var cone = new PieDecoration(2000, 60, lifespan, Colors.LightOrange, 0.2, new PositionConnector(effect.Position)).UsingRotationConnector(new AngleConnector(effect.Rotation.Z + 90));
+                replay.Decorations.Add(cone);
+            }
+        }
+
+        // Enfeebling Miasma - Gas Circles
+        if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerEnfeeblingMiasmaGasMoving, out var miasmaAnimation))
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs:280]: && log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.GreerEnfeeblingMiasmaGasClouds, out var miasmaClouds), no need to do a fetch for each "animation"

File: GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs [hunk 2]
```diff
@@ -137,4 +234,238 @@ internal override void ComputePlayerCombatReplayActors(PlayerActor player, Parse
...
+            foreach (EffectEvent effect in stompTheGrowth)
+            {
+                (long start, long end) lifespan = effect.ComputeLifespan(log, effect.Duration);
+                var circle = new CircleDecoration(radius, lifespan, Colors.LightOrange, 0.2, new PositionConnector(effect.Position));
+                replay.Decorations.AddWithGrowing(circle, effect.Time + effect.Duration);
+            }
+        }
+
+        // Gree doesn't create shockwaves
+        if (target.IsSpecies(ArcDPSEnums.TargetID.Greer))
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs:398]: This should be put outside of this API, the API below does not make use of target nor radius, which makes this section of AddStompTheGrowth non-generic

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Raids/W8/GreerTheBlightbringer.cs:398]: Could be actually put in Environment decorations.

**Comment by Linkaaaaa** [comment]
I've also added the Decima adjustments post patch to this

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**